aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -733,6 +733,15 @@ std::tuple<Tensor, Tensor> _solve_helper_cpu(const Tensor& self, const Tensor& A
 
 // Supports arbitrary batch dimensions for self and A
 std::tuple<Tensor,Tensor> solve(const Tensor& self, const Tensor& A) {
+  TORCH_WARN_ONCE(
+    "torch.solve is deprecated in favor of torch.linalg.solve",
+    "and will be removed in a future PyTorch release.\n",
+    "torch.linalg.solve has its arguments reversed and does not return the LU factorization.\n",
```

Comment:
What should a user do if they want the LU factorization?

Reply:
Sadly we do not have a good solution for this yet, as `torch.lu` is split into several functions (`torch.lu` and `torch.lu_unpack`). I would not worry about this too much, as that return was caused by a wrong design: (t's returning some internal part of the implementation that is not linked to the result (`torch.solve` could be implemented via QR or SVD as well).